### ISSUE TITLE
(bug) Google cloud storage retriever only gets the file in the first attempt

### ISSUE THREAD

**Issue by jferrl** [thomaspoignant/go-feature-flag#310]
### Observed behavior

It was not possible to use an instance of `gcstorageretriever.Retriever{}` since it is only able to get the file on the initial bootstrap of go-feature-flag instance. After feature flag client is up and running, the internal flags cache is no longer updated when **gcstorageretriever** is used.

### Expected Behavior

`gcstorageretriever.Retriever{}` should be able to the flags when the file stored in google cloud storage changes.

### Steps to reproduce

Using the following code(just an exaple): 

```go
ffcli, err := ffclient.New(ffclient.Config{
  PollingInterval: 3 * time.Second,
  Retriever: &gcstorageretriever.Retriever{
      Options: []option.ClientOption{option.WithoutAuthentication()},
  Bucket: "my-bucket",
  Object: "flags.yaml",
  }
})
defer ffcli.Close()
```

After the bootstrap, we are able to get current flags data, for example using the following func:

```go
flags := ffcli.AllFlagsState(user)
```

But, if the file is updated in google cloud storage, go-feature-flag is not able to retrieve the current config.

Moreover, We can see several log traces that refer to the following line of code(saying file not found)
```go
fmt.Errorf("unable to read from GCP Object %s in Bucket %s, error: %s", retriever.Bucket, retriever.Object, err)
```

**Comment by thomaspoignant** [comment]
@jferrl thanks for reporting this issue.

I am sorry that it is not working as expected 😓
I am currently on PTO so I am not able to check this before next week.

In the mean time if you want to work on resolving this issue it will be much appreciated.

**Comment by jferrl** [comment]
> @jferrl thanks for reporting this issue.
> 
> I am sorry that it is not working as expected 😓 I am currently on PTO so I am not able to check this before next week.
> 
> In the mean time if you want to work on resolving this issue it will be much appreciated.

No worries! I'm working on the fix! 
Thanks! 😄

**Comment by thomaspoignant** [comment]
Resolves by #311
